components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -452,93 +452,135 @@ private String getOIDCAuthzEndpoint(Map<String, String> authenticatorProperties)
-            OAuthClientResponse oAuthResponse = getOauthResponse(oAuthClient, accessTokenRequest);
+        Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+        if (StringUtils.isBlank(idToken) && requiredIDToken(authenticatorProperties)) {
+            throw new AuthenticationFailedException(ErrorMessages.ID_TOKEN_MISSED_IN_OIDC_RESPONSE.getCode(),
+                    String.format(ErrorMessages.ID_TOKEN_MISSED_IN_OIDC_RESPONSE.getMessage(),
+                            getTokenEndpoint(authenticatorProperties),
+                            authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_ID)));
+        }
+
+        OIDCStateInfo stateInfoOIDC = new OIDCStateInfo();
+        stateInfoOIDC.setIdTokenHint(idToken);
+        context.setStateInfo(stateInfoOIDC);
 
-            // TODO : return access token and id token to framework
-            String accessToken = oAuthResponse.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN);
+        AuthenticatedUser authenticatedUser;
+        Map<ClaimMapping, String> claims = new HashMap<>();
+        Map<String, Object> jsonObject = new HashMap<>();
 
-            if (StringUtils.isBlank(accessToken)) {
-                throw new AuthenticationFailedException(ErrorMessages.ACCESS_TOKEN_EMPTY_OR_NULL.getCode(),
-                        ErrorMessages.ACCESS_TOKEN_EMPTY_OR_NULL.getMessage());
+        if (StringUtils.isNotBlank(idToken)) {
+            jsonObject = getIdTokenClaims(context, idToken);
+            if (jsonObject == null) {
+                String errorMessage = ErrorMessages.DECODED_JSON_OBJECT_IS_NULL.getMessage();
+                if (log.isDebugEnabled()) {
+                    log.debug(errorMessage);
+                }
+                throw new AuthenticationFailedException(ErrorMessages.DECODED_JSON_OBJECT_IS_NULL.getCode(),
+                        errorMessage);
             }
 
-            String idToken = oAuthResponse.getParam(OIDCAuthenticatorConstants.ID_TOKEN);
-            Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
-            if (StringUtils.isBlank(idToken) && requiredIDToken(authenticatorProperties)) {
-                throw new AuthenticationFailedException(ErrorMessages.ID_TOKEN_MISSED_IN_OIDC_RESPONSE.getCode(),
-                        String.format(ErrorMessages.ID_TOKEN_MISSED_IN_OIDC_RESPONSE.getMessage(),
-                                getTokenEndpoint(authenticatorProperties),
-                                authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_ID)));
+            String idpName = context.getExternalIdP().getIdPName();
+            String sidClaim = (String) jsonObject.get(OIDCAuthenticatorConstants.Claim.SID);
+            if (StringUtils.isNotBlank(sidClaim) && StringUtils.isNotBlank(idpName)) {
+                // Add 'sid' claim into authentication context, to be stored in the UserSessionStore for
+                // single logout.
```

Comment:
Can we use this as a single line comment?

Reply:
Fixed 3b4da7ee38b683b2494433893ed32ce3d9bf1bd3